### PR TITLE
added revocation-list Endpoint to fetch revocked credentials list

### PR DESCRIPTION
**Pull request by Sreejit-K**
Added an endpoint for revocation-list.

### CODE REVIEW

File: api-documentation/issue-api.yaml [hunk 1]
```diff
@@ -184,11 +184,11 @@ paths:
             'appliation/json':
               schema:
                 $ref: '#/components/schemas/verifyResponse'
-  /revocation-list:
+  /credentials/revocation-list:
```
Comment (srprasanna) [inline api-documentation/issue-api.yaml:187]: Isnt the credential part of the base url?

File: services/credentials-service/src/credentials/credentials.controller.ts [hunk 2]
```diff
@@ -116,4 +116,17 @@ export class CredentialsController {
     return this.credentialsService.verifyCredential(credId);
   }
 
+  @Get('revocation-list')
```
Comment (srprasanna) [inline services/credentials-service/src/credentials/credentials.controller.ts:119]: Is this part of the specification. Response Type is missing.

File: services/credentials-service/src/credentials/credentials.service.ts [hunk 3]
```diff
@@ -319,4 +319,44 @@ export class CredentialsService {
       return { id: cred.id, ...signed };
     });
   }
+
+  async getRevocationList(
+    issuerId: string,
+    page = 1,
+    limit= 1000,
+  ){
+    let revocationList: any;
```
Comment (srprasanna) [inline services/credentials-service/src/credentials/credentials.service.ts:328]: Dont use Any. Use explicit Type. This is an array of strings right?

File: services/credentials-service/src/credentials/credentials.service.ts [hunk 4]
```diff
@@ -319,4 +319,44 @@ export class CredentialsService {
...
+        },
+        skip: (page -1) * limit,
+        take: limit,
+        orderBy: {
+          issuanceDate: 'desc',
+        },
+      }); 
+    } catch (error) {
+      this.logger.error('Error fetching RevocationList');
+      throw new InternalServerErrorException('Error fetching revocationList');
```
Comment (srprasanna) [inline services/credentials-service/src/credentials/credentials.service.ts:357]: Which HTTP status code is triggered when an exception occurs.

File: services/credentials-service/src/revocation-list/revocation-list.module.ts [hunk 5]
```diff
@@ -4,7 +4,7 @@ import { RevocationList } from './revocation-list.helper';
 import { PrismaClient } from '@prisma/client';
 import { RevocationListImpl } from './revocation-list.impl';
 import { CredentialsModule } from 'src/credentials/credentials.module';
-import { IdentityUtilsService } from 'src/credentials/utils/identity.utils.service';
+import { IdentityUtilsService } from './../credentials/utils/identity.utils.service';
```
Comment (srprasanna) [inline services/credentials-service/src/revocation-list/revocation-list.module.ts:7]: WHy is line 7 changed but not line 6?

File: api-documentation/issue-api.yaml (hunk 1 above)
Comment (Sreejit-K) [reply api-documentation/issue-api.yaml:187]: No @srprasanna, credentials is not part of the baseURL. We added this endpoint in the credential services so had to add the credentials.

File: services/credentials-service/src/revocation-list/revocation-list.module.ts (hunk 5 above)
Comment (Sreejit-K) [reply services/credentials-service/src/revocation-list/revocation-list.module.ts:7]: Have made changes for like 6 as well

File: services/credentials-service/src/credentials/credentials.service.ts (hunk 4 above)
Comment (Sreejit-K) [reply services/credentials-service/src/credentials/credentials.service.ts:357]: This function throws a 500 status code.

File: services/credentials-service/src/credentials/credentials.service.ts (hunk 3 above)
Comment (Sreejit-K) [reply services/credentials-service/src/credentials/credentials.service.ts:328]: Have made the changes its and array of Objects with the object having the details like DID, status and Issuer.

File: services/credentials-service/src/credentials/credentials.controller.ts (hunk 2 above)
Comment (Sreejit-K) [reply services/credentials-service/src/credentials/credentials.controller.ts:119]: The endpoint is as per the specification. And I have added the Response Type.

**Pull request merged**